dp/drivers/i40e.c

```diff
@@ -327,9 +327,10 @@ static int reta_update(struct ix_rte_eth_dev *dev, struct rte_eth_rss_reta *reta
 
 	/* first convert reta_conf to dpdk format*/
 	for (i = 0; i < dev->data->nb_rx_fgs / 64; i++) {
-		r_reta_conf[i].mask = reta_conf->mask[BITMAP_POS_IDX(i * 64)] & 0xFFFFFFFF;
-		r_reta_conf[i].mask = (reta_conf->mask[BITMAP_POS_IDX(i * 64 + 32)]
-				>> BITMAP_POS_SHIFT(32)) & 0xFFFFFFFF;
+		r_reta_conf[i].mask = reta_conf->mask[BITMAP_POS_IDX(i * 64)] >> BITMAP_POS_SHIFT(i * 64);
+		/* If a long is only 32 bits, need to copy the next element of the bitmap for the high 32 bits */
```

Comment:
We don't support 32-bit platforms so it's safe to assume that a long is 64 bits wide, so it's OK to remove the comment and the if.